File: src/main/java/org/tindalos/principle/infrastructure/plugin/LogPrinter.java
package org.tindalos.principle.infrastructure.plugin;

import org.apache.maven.plugin.logging.Log;
import org.tindalos.principle.app.Printer;

public class LogPrinter implements Printer {

    private final Log log;

    public LogPrinter(Log log) {
        this.log = log;
    }

    @Override
    public void printInfo(String text) {
        log.info(text);
    }

    @Override
    public void printWarning(String text) {
        log.warn(text);
    }
}

File: src/main/java/org/tindalos/principle/infrastructure/plugin/LogPrinter.scala
package org.tindalos.principle.infrastructure.plugin

import org.tindalos.principle.domain.resultprocessing.reporter.Printer
import org.apache.maven.plugin.logging.Log

class LogPrinter(val log:Log) extends Printer {
  
  	def printInfo(text:String) = log.info(text)
  	def printWarning(text:String) = log.warn(text)

}